tugboat/parser_engine/generate_intermediary.py

```diff
@@ -0,0 +1,150 @@
+import yaml
+import re
+
+import netaddr
+from base import ParserEngine
+from utils.excel_parser import ExcelParser
+import utils.settings as settings
+
+
+class GenerateYamlFromExcel(ParserEngine):
+    def __init__(self, file_name, excel_specs):
+        self.HOST_TYPES = settings.HOST_TYPES
+        self.PRIVATE_NETWORK_TYPES = settings.PRIVATE_NETWORK_TYPES
+        self.IPS_TO_LEAVE = settings.IPS_TO_LEAVE
+        parsed_data = self.get_parsed_data(file_name, excel_specs)
+        self.ipmi_data = parsed_data['ipmi_data'][0]
+        self.hostnames = parsed_data['ipmi_data'][1]
+        self.network_data = self.get_network_data(parsed_data['network_data'])
+        self.host_type = {}
+        self.data = {
+            'network': {},
+            'baremetal': {},
+        }
+        self.racks = set()
+
+    def get_parsed_data(self, file_name, excel_specs):
+        parser = ExcelParser(file_name, excel_specs)
+        return parser.get_data()
+
+    def get_network_data(self, raw_data):
```

Comment:
If this is generic function, it can go to parent class. Check similar functions in the rest of the code